### PR TITLE
Misc fixes and tweaks to upload

### PR DESCRIPTION
**Pull request by sampierson**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -8,6 +8,12 @@ Installation
 ------------
 :code:`pip install hca`.
 
+Install the libmagic library.  On Linux machines this library is already present.
```
Comment (kislyuk) [inline README.rst]: I think you should not assume libmagic is installed. For example I'm not sure if it's installed in the Docker Ubuntu distribution by default. Suggest adding the apt-get and yum install lines.

Comment (kislyuk) [reply README.rst]: Also, I'm worried about depending on libmagic. In the past I've seen this and other native library dependencies be really troublesome. Suggest looking into using a pure Python implementation e.g. https://github.com/cdgriffith/puremagic.

File: hca/upload/media_type.py [hunk 2]
```diff
@@ -6,22 +6,60 @@
...
+        r")?"                      # end optional suffix group
+    )
+    _PARAM_REGEX = re.compile(
+        r"\s*"                               # optional whitespace
+        r"(?P<key>"                          # start of group 'key'
+        r"[" + RFC7320_TOKEN_CHARSET + "]+"  # key
+        r")"                                 # end of group 'key'
+        r"\s*=\s*"                           # '='
+        r"(?P<value>"                        # start of group 'value'
+        r'"[^"]*"'                           # Any doublequoted string
```
[the remark this answers is not in the document]
Comment (Bento007) [reply hca/upload/media_type.py]: it's allowed in r-strings

File: README.rst (hunk 1 above)
Comment (ttung) [reply README.rst]: If this were the server code, I'd be okay with this.  However, end users will run into this and not everyone is comfortable with homebrew.

Comment (sampierson) [reply README.rst]: I switched to puremagic.

File: hca/upload/media_type.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (sampierson) [reply hca/upload/media_type.py]: @Bento007 he's talking about whether \" is allowed inside quoted-string params of media-types.

@ttung Technically, yes \" (a "quoted-pair" per https://tools.ietf.org/html/rfc7230#section-3.2.6) is allowed.  You really love these technically-correct hyper-obscure corner cases, don't you :)
I'll fix this later.  Right now my priority is getting the DCP integrated, not wasting time implementing code that will never be used.  If you can show me a case where someone has used a quoted-pair on a media-type parameter I'll bump up the priority of this, but frankly even finding cases where people use params is hard.

File: hca/upload/media_type.py [hunk 3]
```diff
@@ -6,22 +6,60 @@
...
     @classmethod
     def from_string(cls, media_type):
-        split_on_semicolons = media_type.split(';')
-        type_info = split_on_semicolons.pop(0)
-        type_info = re.match("(?P<top_level_type>.+)/(?P<subtype>[^+]+)(\+(?P<suffix>.+))?", type_info)
-        parameters = {}
-        for parameter in split_on_semicolons:
-            param_match = re.match('(.+)=(.+)', parameter.strip())
-            parameters[param_match.group(1)] = param_match.group(2).strip('"')
+        first_semi_position = media_type.find(';')
```
[the remark this answers is not in the document]
Comment (sampierson) [reply hca/upload/media_type.py]: The rules for cookies are slightly different.  Yes that code could would probably do the job for the simple cases we have to handle in the DCP.  But as you're being super-pedantic about handling obscure corner cases, then no, it's not an exact match, so we have to do something more specific.  I have to say however that we are waaay past the point of diminishing returns now.

File: hca/upload/media_type.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ttung) [reply hca/upload/media_type.py]: I think I mentioned this in the earlier PR -- while I don't think it's required that we build a perfect parser, we should definitely screen for known issues where we _produce_ the k-v pairs.  i.e., panic and throw an exception if someone tries to put double-quotes https://github.com/HumanCellAtlas/dcp-cli/blob/71f33dab26590acedc517319ded63b558174f128/hca/upload/media_type.py#L30

**Pull request merged**